### PR TITLE
tests: rename addFilter() to prependFilter()

### PR DESCRIPTION
**Pull request by rgs1**
To avoid confusion when writing integration tests, make it clear
that filters are prepended.

It's currently an addition not an actual rename, because the rename would
break the envoy-filter-example repo and the ASAN/TSAN tests rely on 
that. So I'll get rid of addFilter() after this lands and the example repo
is updated. 

Signed-off-by: Raul Gutierrez Segales <rgs@pinterest.com>

### CODE REVIEW

File: test/integration/multiplexed_integration_test.h [hunk 1]
```diff
@@ -16,10 +16,10 @@ class Http2IntegrationTest : public HttpProtocolIntegrationTest {
   void simultaneousRequest(int32_t request1_bytes, int32_t request2_bytes);
 
 protected:
-  // Utility function to add filters.
-  void addFilters(std::vector<std::string> filters) {
+  // Utility function to prepend filters.
```
Comment (rojkov) [inline test/integration/multiplexed_integration_test.h:19]: nit: I guess we could mention here that the filters are going to be prepended in reverse order.

**Review by rojkov — COMMENTED** [review]
Thanks! This way it's clearer indeed.

**Review by yanavlasov — APPROVED** [review]

**Comment by rgs1** [comment]
Counterpart:

https://github.com/envoyproxy/envoy-filter-example/pull/151

**Pull request merged**